status/consumers.py

```diff
@@ -0,0 +1,148 @@
+            return
+
+        user, user_id = _find_user(access_token)
+        kwargs = self.scope["url_route"]["kwargs"]
+
+        self.room_group_name = kwargs.get('workspace_hashed_value')
+
+        async_to_sync(self.channel_layer.group_add)(
+            self.room_group_name,
+            self.channel_name
+        )
+
+        # Check workspace `hashed_value` is valid.
+        try:
+            w: Workspace = Workspace.objects.prefetch_related('members').get(
+                hashed_value=kwargs.get('workspace_hashed_value'))
+            all_users_in_workspace: [User] = w.members.all()  # Have to find better way to find a user.
+        except Workspace.DoesNotExist as e:
+            print(f'{e}')
+            return
+
+        if user not in all_users_in_workspace:
+            print(f'{user} is not in workspace')
+            return
+
+        self.accept()
+        self.send_json(content={'user_id': user_id})
+
+    message_field = ['status_message', 'status_icon', 'until']
+
+    def receive_json(self, content, **kwargs):
+        """
+        Find user model using access token.
+        And check what fields are unfilled.
+        If no problem, Send message to everyone.
+        """
+        access_token = _find_access_token(self.scope)  # Don't check about access token.
+        user, _ = _find_user(access_token)  # May return None if access token is expired.
+        if user is None:
+            self.send_json(content={'msg': 'access token is expired or invalid.'}, close=True)
+
+        data = {'type': 'status.broadcast'}
+        not_filled = []
+        for field in self.message_field:
+            if content.get(field, None) is None:
+                not_filled.append(field)
+
+        if len(not_filled):
+            self.send_json(content={'msg': f'Not filled this fields: {not_filled}'})
+        else:
+            try:
+                status = UserStatus.objects.get(user=user)
+                status.message = content.get('status_message')
+                status.icon = content.get('status_icon')
+                status.until = content.get('until')
+                status.save()
+                del status
+            except UserStatus.DoesNotExist:
+                workspace = Workspace.objects.get(hashed_value__exact=self.room_group_name)
+                UserStatus.objects.create(
+                    message=content.get('status_message'),
+                    icon=content.get('status_icon'),
+                    until=content.get('until'),
+                    workspace=workspace,
+                    user=user
+                )
+                del workspace
+
+            # Refine data.
+            print(f'{datetime.utcnow()=}')
+            status = UserStatus.objects.prefetch_related('workspace') \
+                .filter(Q(workspace__hashed_value__exact=self.room_group_name) &
+                        Q(until__gt=datetime.utcnow()))
+
+            result = []
+            for s in status:
```

Comment:
Would you change UserStatus  status with user_status and  s as status?
or it's better too use s?

Reply:
Or if the value will be used temporarily, and the life cycle is short,
can use single word?

Reply:
https://github.com/Team-Discipline/Xlack-Backend/pull/48/commits/f6f2f8b7dc08da9e5fcd674bf3344a0474185799

Fixed.

Reply:
that's the comfort